### DISCUSSION TITLE
Using 16 MB of Flash on the Raspberry Pi Pico board with PlatformIO

### DISCUSSION THREAD

**Discussion by MicSG-dev** [earlephilhower/arduino-pico#1763] [Show and tell]
After some considerable time trying to use my Raspberry Pi Pico which has 16 MB of Flash along with PlatformIO, I managed to figure out how to do this. Hope this can help someone.

1. In your project's platformio.ini file, use the "Generic" board (list of boards and flash sizes available [here](https://github.com/maxgerhardt/platform-raspberrypi/tree/develop/boards) ): `board = Generic`;
2. Set the Flash memory size to 16 MB: `board_build.flash_size = 16MB`;
3. Set the memory size allocated to the FileSystem (FS) to something less than 16 MB, (if, for example, the value 15 MB is entered for FS, 1 MB will be allocated to Sketch. For more details, see the [documentation]( https://arduino-pico.readthedocs.io/en/latest/platformio.html#flash-size)) : `board_build.filesystem_size = 4mb`;
![image](https://github.com/earlephilhower/arduino-pico/assets/71986598/bd750a4c-c01d-4921-beab-53e71439b2e1)

**Comment by MicSG-dev** [comment]
For convenience, copy and paste:
```
[env:pico]
platform = https://github.com/maxgerhardt/platform-raspberrypi.git
board = Generic
framework = arduino
board_build.core = earlephilhower
board_build.flash_size = 16MB ; Flash Size: 16MB
board_build.filesystem_size = 4mb ; Sketch: 12MB, FS: 4MB
```

**Comment by jaco-pixeldump** [comment]
board = generic ; lowercase

* set sizes in MB (UPPERCASE)
